pkg/flare/archive.go

```diff
@@ -694,6 +700,22 @@ func zipDiagnose(tempDir, hostname string) error {
 	return writeScrubbedFile(f, b.Bytes())
 }
 
+func zipDatadogConnectivity(tempDir, hostname string) error {
```

Comment:
Bonus: The code of this function is very similar to several other functions like `zipClusterAgentDiagnose`, `zipSecrets` or `zipDiagnose`. You can factorize them.